contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -220,6 +220,19 @@ func (t *T) AcceptOffer(offer *mesos.Offer) bool {
 		return false
 	}
 
+	// check the NodeSelector
+	labels := map[string]string{}
+	for _, a := range offer.Attributes {
+		if a.GetType() == mesos.Value_TEXT {
+			labels[*a.Name] = *a.GetText().Value
```

Comment:
should use the getters instead of pointer de-referencing.
do we need to worry about validation here?
```go
name := a.GetName()
if name == "" { continue }
value := a.GetText().GetValue()
// is an empty value ok?
```